lib/msf/core/modules/metadata/obj.rb

```diff
@@ -49,7 +50,7 @@ def initialize(module_instance, obj_hash = nil)
     sort_platform_string
 
     @arch = module_instance.arch_to_s
-    @rport = module_instance.datastore['RPORT'].to_s
```

Comment:
Doesn't seem to impact your changes is there a reason to remove the `.to_s` here?  This could could impact at the very least the values in the cached json.

`"RPORT": "45"` may become `"RPORT": 45`

Reply:
Yep, noted in PR description. `RPORT` should be an int, not a string. Could lead to breakage, but we'd better not cater to what's incorrect. I'm thinking search will break, so I'll go fix that now.

Reply:
Thanks!